scripts/deploy_subnet/deploy.sh

```diff
@@ -780,44 +783,66 @@ if [[ $local_deploy = true ]]; then
   # with validator accounts. So, we deploy the bucket and credit manager contracts via the last evm
   # account. Ideally, we would use validator 0 and somehow avoid nonce clashes.
   deployer_pk=$(jq .[9].private_key < "${IPC_CONFIG_FOLDER}"/evm_keystore.json | tr -d '"')
   cd "${IPC_FOLDER}/recall-contracts"
-  echo "$DASHES deploy bucket and credit manager output $DASHES"
+  echo "$DASHES Deploy blob, bucket, and credit manager contracts $DASHES"
   deploy_blob_manager_token_out="$(forge script script/BlobManager.s.sol --private-key "${deployer_pk}" --rpc-url http://localhost:"${ETHAPI_HOST_PORTS[0]}" --tc DeployScript --sig 'run()' --broadcast --timeout 120 -g 100000 -vv)"
   deploy_bucket_manager_token_out="$(forge script script/BucketManager.s.sol --private-key "${deployer_pk}" --rpc-url http://localhost:"${ETHAPI_HOST_PORTS[0]}" --tc DeployScript --sig 'run()' --broadcast --timeout 120 -g 100000 -vv)"
   deploy_credit_manager_token_out="$(forge script script/CreditManager.s.sol --private-key "${deployer_pk}" --rpc-url http://localhost:"${ETHAPI_HOST_PORTS[0]}" --tc DeployScript --sig 'run()' --broadcast --timeout 120 -g 100000 -vv)"
   echo ""
+  echo "Blob manager output:"
+  echo "$deploy_blob_manager_token_out"
+  echo "Bucket manager output:"
   echo "$deploy_bucket_manager_token_out"
+  echo "Credit manager output:"
   echo "$deploy_credit_manager_token_out"
   echo ""
-  # note: these are consistently going to be 0xe1Aa25618fA0c7A1CFDab5d6B456af611873b629 and
-  # 0xf7Cd8fa9b94DB2Aa972023b379c7f72c65E4De9D, respectively, for localnet
+  # note: these are consistently going to be the following addresses on localnet:
+  # - Blob: 0x8ce361602B935680E8DeC218b820ff5056BeB7af
+  # - Bucket: 0xeD1DB453C3156Ff3155a97AD217b3087D5Dc5f6E
+  # - Credit: 0x196dBCBb54b8ec4958c959D8949EBFE87aC2Aaaf
   BLOB_MANAGER_ADDRESS=$(echo "$deploy_blob_manager_token_out" | sed -n 's/.*contract BlobManager *\([^ ]*\).*/\1/p')
   BUCKET_MANAGER_ADDRESS=$(echo "$deploy_bucket_manager_token_out" | sed -n 's/.*contract BucketManager *\([^ ]*\).*/\1/p')
   CREDIT_MANAGER_ADDRESS=$(echo "$deploy_credit_manager_token_out" | sed -n 's/.*contract CreditManager *\([^ ]*\).*/\1/p')
   echo "Blob manager address: ${BLOB_MANAGER_ADDRESS}"
   echo "Bucket manager address: ${BUCKET_MANAGER_ADDRESS}"
   echo "Credit manager address: ${CREDIT_MANAGER_ADDRESS}"
   echo
+
+  # buy 5000 credits for each account
+  echo "Buying credits for test accounts..."
+  credit_amount="5000"
+  max_retries=5
+  for i in {0..9}
+  do
+    private_key=$(jq .["$i"].private_key < "${IPC_CONFIG_FOLDER}"/evm_keystore.json | tr -d '"')
+    attempt=1
+    # due to potential nonce race conditions with the first three accounts, we need to retry the
+    # transaction until it succeeds—else, we'll continue and simply skip the account since it's not
+    # recommended to use the validator accounts during development, anyways
```

Comment:
this rarely happens, but if there's a nonce race condition with the validator accounts, we use some simple retry logic. else, the script would exit early.